rails_app/app/interactors/send_newsletter.rb

```diff
@@ -12,7 +12,9 @@ def perform
   end
 
   def call
-    User.all.each do |user|
+
+    User
+    User.in_batches.each do |user|
```

Comment:
You can rely on `find_each` for this.

https://api.rubyonrails.org/v7.0.4.2/classes/ActiveRecord/Batches.html#method-i-find_each

Reply:
Yep, I ran a quick console test and it doesn't work as I would expect 😅
I'll submit the tested version soon.

Reply:
`find_each` is not used, marking this as solved.